gnat2goto/driver/tree_walk.adb

```diff
@@ -1610,6 +1627,56 @@ package body Tree_Walk is
       return Ret;
    end Do_If_Statement;
 
+   ------------------------------------
+   -- Do_Incomplete_Type_Declaration --
+   ------------------------------------
+
+   procedure Do_Incomplete_Type_Declaration (N : Node_Id) is
+      Entity : constant Entity_Id := Defining_Identifier (N);
+      --  Only complete types should be inserted in the symbol table.
+      --  If an incomplete type declaration is inserted it will prevent
+      --  the full declaration being entered into the symbol talble.
+      --
+      --  The full view of an incomplete_type_declaration is obtained
+      --  by calling the Full_View function.  As the compiler has completed
+      --  semantic analysis before invoking the gnat to goto translation
+      --  all incomplete_type_declarations should have a full view.
+      Full_View_Entity : constant Entity_Id := Full_View (Entity);
+   begin
+      if Is_Incomplete_Type (Entity) then
+         --   If the incomplete_type_declaration is completed by a
+         --   private_type_declaration, the private_type_declaration
+         --   has to be processed to obtain the full view of the type.
+         if not Is_Private_Type (Full_View_Entity) then
+            if Nkind (Declaration_Node (Full_View_Entity)) =
+              N_Full_Type_Declaration
+            then
+               --  The full_type_declaration corresponding to the
+               --  incomplete_type_declaration is Full_View_Entity
+               --  register the full view in the symbol_table.
+               Register_Type_Declaration
+                 (Declaration_Node (Full_View_Entity), Full_View_Entity);
+            else
+               Report_Unhandled_Node_Empty
```

Comment:
We originally intended the `Report_Unhandled_Node` facilities for unsupported feature, i.e. things that GNAT would compile but gnat2goto does not support (yet). Here it seems more like something that would not be a valid Ada code.

Reply:
It would be a fault in the front-end if this occurred.  How should we abort and give a message in this situation?